pkg/nullio/content_type.go

```diff
@@ -0,0 +1,43 @@
+package nullio
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// ContentTypeWriter implements io.Writer.
+// It forwards the first line as Content Type Header and streams the passed data to the Target.
+type ContentTypeWriter struct {
+	Target       http.ResponseWriter
+	mimeTypeSent bool
+	mimeType     []byte
+}
+
+func (w *ContentTypeWriter) Write(p []byte) (count int, err error) {
+	if w.mimeTypeSent {
+		count, err = w.Target.Write(p)
+		return count, fmt.Errorf("could not write to target: %w", err)
+	}
+
+	for i, char := range p {
+		if char != '\n' {
+			continue
+		}
+
+		w.mimeType = append(w.mimeType, p[:i]...)
+		w.Target.Header().Set("Content-Type", string(w.mimeType))
```

Comment:
We should also consider to add a [`Content-Disposition` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) with the `attachment` keyword (this is a must), and optionally provide the filename in the header as well.

Reply:
Yeah, this would be an easy improvement :+1: 
More complex would be to set the `Content-Length` header for large files. For small files this is done automatically by Go:
``` Go
        // Additionally, if the total size of all written
	// data is under a few KB and there are no Flush calls, the
	// Content-Length header is added automatically.
```
For larger files, we could think about adding the Content-Length header the way [e1d16e5](https://github.com/openHPI/poseidon/pull/159/commits/e1d16e523b5c7bafc8485b843836388110eca253) has added (unnecessarily) the Content-Type?!

Reply:
Sure - if you want, just do so. We already have the infrastructure anyway with your implementation and just need to swap the command and header 😁